### PR TITLE
Refactor: Fix unused imports and add ExplicitResultTypes

### PR DESCRIPTION
**Pull request by arunkumar461**

### CODE REVIEW

File: app/models/repository/Merger.scala [hunk 1]
```diff
@@ -52,7 +51,7 @@ object Merger {
               jointlyLetYesOrNo = jointlyLet,
               totalIncomeAmount = ukOtherProperty.income.flatMap(_.ukOtherRentARoom.map(_.rentsReceived)).getOrElse(0),
               claimExpensesOrRRR = ClaimExpensesOrRRR(
-                extracted.fold(!amountClaimedMaybe.isEmpty)(_.claimExpensesOrRRR),
+                extracted.fold(amountClaimedMaybe.isDefined)(_.claimExpensesOrRRR),
```
Comment (modivilioglu) [inline app/models/repository/Merger.scala:54]: Nice

File: app/models/repository/Merger.scala [hunk 2]
```diff
@@ -292,7 +298,7 @@ object Merger {
                 renovationAllowanceBalancingChargeYesNo = extractedMaybe
                   .map(_.renovationAllowanceBalancingChargeYesNo)
                   .getOrElse(
-                    !fromDownstreamAdjustment.businessPremisesRenovationAllowanceBalancingCharges.isEmpty
+                    fromDownstreamAdjustment.businessPremisesRenovationAllowanceBalancingCharges.isDefined
```
Comment (modivilioglu) [inline app/models/repository/Merger.scala:301]: nice

File: app/models/responses/PropertyAnnualSubmission.scala [hunk 3]
```diff
@@ -291,8 +289,7 @@ object PropertyAnnualSubmission {
     val electricChargePointAllowanceLens = GenLens[UkOtherAllowances](_.electricChargePointAllowance)
 
     // Focuses
-    val focusFromRequestOnToannualInvestmentAllowanceLens =
-      ukOtherPropertyLens.andThen(ukOtherAllowancesLens).andThen(annualInvestmentAllowanceLens)
+    ukOtherPropertyLens.andThen(ukOtherAllowancesLens).andThen(annualInvestmentAllowanceLens)
```
Comment (modivilioglu) [inline app/models/responses/PropertyAnnualSubmission.scala:292]: "val focusFromRequestOnToannualInvestmentAllowanceLens =" deleted?

File: app/services/PropertyService.scala [hunk 4]
```diff
@@ -661,12 +639,10 @@ class PropertyService @Inject() (
             ctx.nino.value,
             ctx.incomeSourceId.value
           )
-          .leftFlatMap(e =>
-            e match {
-              case DataNotFoundError => ITPEnvelope.liftPure(emptyPropertyAnnualSubmission)
-              case _                 => ITPEnvelope.liftEither(e.asLeft[PropertyAnnualSubmission])
-            }
-          )
+          .leftFlatMap {
```
Comment (modivilioglu) [inline app/services/PropertyService.scala:642]: nice

File: app/models/responses/PropertyAnnualSubmission.scala (hunk 3 above)
Comment (modivilioglu) [reply app/models/responses/PropertyAnnualSubmission.scala:292]: if it is not used, the assigned line ```ukOtherPropertyLens.andThen(ukOtherAllowancesLens).andThen(annualInvestmentAllowanceLens``` should be deleted as well.

Comment (arunkumar461) [reply app/models/responses/PropertyAnnualSubmission.scala]: Scala fix plugin removes only variable its funny, removed the entire line now. Thanks for pointing it.

**Review by modivilioglu — APPROVED** [review]
LGTM. Nice

**Pull request merged**